src/ui/molecules/stripe-express-checkout-element.svelte

```diff
@@ -0,0 +1,103 @@
+<script lang="ts">
+  import type {
+    StripeError,
+    StripeElements,
+    StripeExpressCheckoutElement,
+    StripeExpressCheckoutElementConfirmEvent,
+    StripeExpressCheckoutElementReadyEvent,
+    StripeExpressCheckoutElementClickEvent,
+  } from "@stripe/stripe-js";
+  import {
+    StripeService,
+    StripeServiceError,
+  } from "../../stripe/stripe-service";
+
+  import { getContext, onDestroy, onMount } from "svelte";
+  import TextSeparator from "../atoms/text-separator.svelte";
+  import { LocalizationKeys } from "../localization/supportedLanguages";
+  import { Translator } from "../localization/translator";
+  import { Writable } from "svelte/store";
+  import { translatorContextKey } from "../localization/constants";
+
+  export interface Props {
+    onError: (error: StripeServiceError) => void | Promise<void>;
+    onReady: () => void | Promise<void>;
+    onSubmit: (
+      paymentMethod: string,
+      emailValue: string,
+    ) => void | Promise<void>;
+    elements: StripeElements;
+    billingAddressRequired: boolean;
+  }
+
+  const {
+    onError,
+    onReady,
+    onSubmit,
+    elements,
+    billingAddressRequired,
+  }: Props = $props();
+
+  const translator = getContext<Writable<Translator>>(translatorContextKey);
+
+  let expressCheckoutElement: StripeExpressCheckoutElement | null = null;
+  let hideExpressCheckoutElement = $state(false);
+  const expressCheckoutElementId = "express-checkout-element";
+
+  const onLoadErrorCallback = async (event: {
+    elementType: "expressCheckout";
+    error: StripeError;
+  }) => {
+    await onError(StripeService.mapInitializationError(event.error));
+  };
+
+  const onConfirmCallback = async (
+    event: StripeExpressCheckoutElementConfirmEvent,
+  ) => {
+    onSubmit(event.expressPaymentType, event.billingDetails?.email ?? "");
+  };
+
+  const onReadyCallback = async (
+    event: StripeExpressCheckoutElementReadyEvent,
+  ) => {
+    hideExpressCheckoutElement = !event.availablePaymentMethods;
+    onReady();
```

Comment:
```suggestion
  try {
    await onReady();
  } catch (err) {
    onError(err as StripeServiceError);
  }
``` ?

Reply:
I don't think this should be needed. An error inside the `onReady()` method should be handled by the parent component